lib/instance_summary.rb

```diff
@@ -0,0 +1,287 @@
+require 'text-table'
+require 'thread'
+require 'thwait'
+
+module BushSlicer
+  class InstanceSummary
+    attr_accessor :jenkins
+
+    def initialize(jenkins: nil)
```

Comment:
you allow initialization without a jenkins instance. But later you call methods on `jenkins` without checking for `nil`. So perhaps `: nil` part needs to be removed. To make sure objects are always created with a `jenkins` object set.